nemo_curator/utils/distributed_utils.py

```diff
@@ -267,6 +269,23 @@ def _set_torch_to_use_rmm():
     torch.cuda.memory.change_current_allocator(rmm_torch_allocator)
 
 
+def select_columns(
+    df: Union[dd.DataFrame, dask_cudf.DataFrame],
+    columns: List[str],
+    filetype: Literal["jsonl", "json", "parquet"],
+    add_filename: bool,
+) -> Union[dd.DataFrame, dask_cudf.DataFrame]:
+    # We exclude parquet because the parquet readers already support column selection
```

Comment:
@sarahyurick I'd love a third eye here, as the tests `test_read_data_select_columns` are failing when using `read_data_fpp`

(When add_filename=True)

Reply:
Wanted to clarify what the goal is here: if `add_filename=True`, then a "filename" column is added to the DataFrame. Then the user prunes to only include `columns` (and the user's `columns` does not "filename"). Is the desired behavior to include the "filename" column anyway or not?

Reply:
The desired behavior would be to include `filename`, as the user might just do `add_filename=True` along with set of columns that doesn't have `filename` as part. 

Assuming df has `id` and `text`
| add_filename | columns | output |
|-----------|--------|--------------|
False | None | ['id', 'text'] |
True | None | ['filename', 'id', 'text'] |
False | ['id'] | ['id']
True | ['id'] | ['filename', 'id'] |
False | ['filename', 'id'] | KeyError
True | ['filename', 'id'] | ['filename', 'id']